### PR TITLE
Output for ,setting up linters ,reflecting on the learning experience,checking repository health ,solving the problem, writing test,using design pattern, setting up software development environment

### PR DESCRIPTION
**Pull request by omokaro-faith**
What does this PR do?

Create output for setting up linters
Creating output for reflecting on the learning experience
Create output for checking repository health
Create output for solving the problem
Create output for writing test
Create output for using design pattern
Create output for setting up software development environment

Description of Task to be completed?
Create output for objective 6

How should this be manually tested?

Navigate to the output folder then view files

Any background context you want to provide?

NA

What are the relevant pivotal tracker stories?

NA

Screenshots (if appropriate)

NA

Questions:

NA

### CODE REVIEW

File: 06-building-a-program-start-to-finish/automating-tasks/Output/README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#### Output
+- **Using**: Gulp
+- **Deliver**: a screencast showing gulp watching and refreshing your browser when changes are made to your HTML, CSS & JS files
+- **With**: *independently*
+
+#### Targeted Learning Outcomes
+[link to screencast showing gulp watching HTML, CSS & JS files !](https://www.youtube.com/watch?v=QxiJnVfEKkQ)
+
+### Author
+### Omokaro Faith
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/automating-tasks/Output/README.md:10]: Cool

File: 06-building-a-program-start-to-finish/checking-repository-health/Output/README.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#### Output
+- **Using**: `Code Climate` and `Coveralls`
+- **Deliver**: a screenshot of your code coverage and GPA badges in your repo's README file
+- **With**: *independently*
+
+#### Targeted Learning Outcomes
+[link to travis and coverage badge](https://github.com/andela-fomokaro/Inverted-Index-Project/blob/develop/Pictures%20of%20eslint%20error%20violation%20and%20houndci/travis%20and%20coverage%20badges.png)
+
+### Author
+### Omokaro Faith
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/checking-repository-health/Output/README.md:10]: Cool

File: 06-building-a-program-start-to-finish/delivering-your-product/Output/README.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#### Output
+- **Using**: Google Doc
+- **Deliver**: `Release notes` for your final product in your repo's wiki
+- **With**: *independently*
+
+#### Targeted Learning Outcomes
+[Link to release note!](https://docs.google.com/a/andela.com/document/d/1kbvLKewvndFRu_7ez4qlVHpyv6Kun1CzDr0X-WlOiuw/edit?usp=sharing)
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/delivering-your-product/Output/README.md:7]: You will also want to add the version of this release

File: 06-building-a-program-start-to-finish/reflecting-on-the-learning-experience/Output/README.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+- **Deliver**: a post reflecting on your learning experience while producing these outputs
+- **With**: *independently*
+
+#### Targeted Learning Outcomes
+#### Output
+
+[Link to Medium Post!](https://medium.com/@faith.omokaro/checkpoints-and-simulations-my-road-to-learning-38c3ccf07c09#.e27qrpifq)
+
+## Author
+### Omokaro Faith
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/reflecting-on-the-learning-experience/Output/README.md:12]: Seen

File: 06-building-a-program-start-to-finish/setting-up-continuous-integration-(build-server)/Output/README.md [hunk 5]
```diff
@@ -0,0 +1,8 @@
+#### Output
+- **Using**: TravisCI
+- **Deliver**: a link to a successful TravisCI build of your `development` branch (it should include test runs)
+- **With**: *independently*
+
+#### Targeted Learning Outcomes
+Link to Successful travis build
+https://travis-ci.org/andela-fomokaro/Inverted-Index-Project
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/setting-up-continuous-integration-(build-server)/Output/README.md:8]: Cool

File: 06-building-a-program-start-to-finish/setting-up-linters/Output/README.md [hunk 6]
```diff
@@ -0,0 +1,8 @@
+#### Output
+- **Using**: Terminal & GitHub
+- **Deliver**: two screenshots of ESLint [Airbnb](https://github.com/airbnb/javascript)  style violations in the terminal and [HoundCI](https://houndci.com/) comment on a PR
+- **With**: *independently*
+
+#### Targeted Learning Outcomes
+ [link to Eslint violation for Houndci](https://github.com/andela-fomokaro/Inverted-Index-Project/blob/features/Pictures%20of%20eslint%20error%20violation%20and%20houndci/Screen%20Shot%202017-03-26%20at%207.43.05%20AM.png) style and
+ [link to Eslint violation for Airbnb](https://github.com/andela-fomokaro/Inverted-Index-Project/blob/features/Pictures%20of%20eslint%20error%20violation%20and%20houndci/Screen%20Shot%202017-03-26%20at%207.20.53%20AM.png) outputs
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/setting-up-linters/Output/README.md:8]: Cool

File: 06-building-a-program-start-to-finish/setting-up-software-development-environments-(deployment)/Output/README.md [hunk 7]
```diff
@@ -0,0 +1,7 @@
+#### Output
+- **Using**: Heroku
+- **Deliver**: live URLs to a `review` app, your `staging` & `production` apps
+- **With**: *independently*
+
+#### Targeted Learning Outcomes
+[Live URL to deployment on production application](https://fomokaro-inverted-production.herokuapp.com/) and [link to heroku application deployment on staging](https://fomokaro-invertedindex-staging.herokuapp.com/)
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/setting-up-software-development-environments-(deployment)/Output/README.md:7]: Why is there not live url to your review app?

File: 06-building-a-program-start-to-finish/setting-up-your-repository/Output/README.md [hunk 8]
```diff
@@ -0,0 +1,7 @@
+OUTPUT
+
+Link to Github repo
+https://github.com/andela-fomokaro/Inverted-Index-Project
+
+## Author
+### Omokaro Faith
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/setting-up-your-repository/Output/README.md:7]: Good

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js [hunk 9]
```diff
@@ -0,0 +1,186 @@
+/**
+ * Implementation of the inverted index data structure.
+ * @author Omokaro Faith <faith.omokaro@andela.com>
+*/
+class Index {
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js:5]: Would it be appropriate to name this class InvertedIndex?

File: 06-building-a-program-start-to-finish/writing-tests/Output/spec/index-spec.js [hunk 10]
```diff
@@ -0,0 +1,163 @@
+const Index = require('../Model/inverted-index');
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/Output/spec/index-spec.js:1]: Please follow the proper naming convention for this files?

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js [hunk 11]
```diff
@@ -0,0 +1,186 @@
...
+    this.index = {};
+    this.temp_search = [];
+    this.searchResult = {};
+    this.allIndex = {};
+  }
+/**
+     * It returns the words passed in small letters and spaces removed
+     * @method tokenize
+     * @param {string} words
+     * @return {String}
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js:20]: Missing JSDoc "return" description.

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js [hunk 12]
```diff
@@ -0,0 +1,186 @@
...
+   * @param {Function} callback - The callback function on file read
+   * @returns {void}
+   */
+  readFile(file, callback) {
+    const reader = new FileReader();
+    reader.onload = callback;
+    reader.readAsText(file);
+  }
+/**
+   * It validates the file passed and returns the error message if an error occurs
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js:42]: This line exceeds the maximum line length of 80. Which is an eslint violation. Please fix

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js [hunk 13]
```diff
@@ -0,0 +1,186 @@
...
+   *
+   * @method validateFile
+   * @param {String} file
+   *
+   * @return {object} validates the file passed
+   */
+  validateFile(file) {
+    const jsonFile = file;
+    let check = {
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js:51]: Where in this method are you using "check". It is defined, reassigned but not used in any way.

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js [hunk 14]
```diff
@@ -0,0 +1,186 @@
...
+        }
+      });
+    } catch (error) {
+      return error.msg;
+    }
+  }
+/**
+     * It returns the index of the file passed
+     * @method createIndex
+     * @param {Object, Object} fileName jsonObject
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js:77]: What is this please?

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js [hunk 15]
```diff
@@ -0,0 +1,186 @@
...
+     * @return {number}
+     */
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js:79]: This JSDoc is not properly done.

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js [hunk 16]
```diff
@@ -0,0 +1,186 @@
...
+  createIndex(fileName, jsonObject) {
+    const newIndex = {};
+    this.validateFile(jsonObject);
+
+    jsonObject.forEach((object, position) => {
+      const longSentence = `${object.title} ${object.text}`;
+      const tokenized = this.tokenize(longSentence);
+      const wordArray = tokenized.split(' ');
+
+      wordArray.forEach((word, pos) => {
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js:89]: Where is **pos** used in this _forEach_ block

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js [hunk 17]
```diff
@@ -0,0 +1,186 @@
...
+  }
+/**
+     * It returns the index of the words
+     * @method getIndex
+     * @param {Object} fileName
+     *
+     * @return {object} gets the index
+     */
+  getIndex(fileName) {
+    if (Object.keys(arguments).length < 1) {
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js:107]: Is this a correct us of the _**arguments**_ variable?

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js [hunk 18]
```diff
@@ -0,0 +1,186 @@
...
+   * @method flattenSearch
+   * @param {void}
+   *
+   * @return {void}
+   */
+  flattenSearch() {
+    this.temp_search = [];
+    for (const arg of arguments) {
+      if (arg instanceof Object && typeof arg !== 'string') {
+        for (const item in arg) {
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js:125]: Using 'ForOfStatement' is not allowed. Please see link your Airbnb style guide

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js [hunk 19]
```diff
@@ -0,0 +1,186 @@
...
+    return this.index[fileName];
+  }
+  /**
+   *
+   *
+   * @method flattenSearch
+   * @param {void}
+   *
+   * @return {void}
+   */
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js:120]: This JSDoc is not right. Please revisit.

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js [hunk 20]
```diff
@@ -0,0 +1,186 @@
...
+  flattenSearch() {
+    this.temp_search = [];
+    for (const arg of arguments) {
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js:123]: Is this the correct use of the _**arguments**_ variable

Using 'ForOfStatement' is not allowed. Please see link your Airbnb style guide

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js [hunk 21]
```diff
@@ -0,0 +1,186 @@
...
+        });
+      }
+    }
+  }
+/**
+     * it return the word searched for in the object it was found
+     *
+     * @method readFile
+     * @param {String} sTerms
+     * @param {String} allIndex
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js:143]: Are you using 'allIndex' anywhere in this method? Eslint flagged this error here - _Expected JSDoc for 'fileName' but found 'allIndex'._

**Review by andela-moe — COMMENTED** [review]
Please address comments

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js (hunk 9 above)
Comment (omokaro-faith) [reply 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js:5]: ok

**Pull request closed without merging**